ambry-api/src/test/java/com.github.ambry/rest/MockRestResponseChannel.java

```diff
@@ -183,12 +183,39 @@ public synchronized void setStatus(ResponseStatus status)
   }
 
   @Override
+  public synchronized ResponseStatus getStatus() {
```

Comment:
It is meant to be thread safe because the interface demands it but I don't see any opportunities for race conditions here. Will remove the `synchronized` keyword.